app/server/src/features/accounts/methods.test.ts

```diff
@@ -0,0 +1,349 @@
+import bcrypt from 'bcryptjs';
+import { register, findUserById, findOrgsByUserId, registerSelf, loginWithPassword } from './methods';
+import { prismaMock } from '../../../mocks/prisma/singleton';
+import * as jwt from '@local/lib/jwt';
+import { toGlobalId } from '../utils';
+import { faker } from '@faker-js/faker';
+
+jest.mock('@local/lib/jwt', () => ({
+    ...jest.requireActual('@local/lib/jwt'), // import and retain the original functionalities
+}));
+
+const toUserId = toGlobalId('User');
+const userData = {
+    email: faker.internet.email(),
+    firstName: faker.name.firstName(),
+    lastName: faker.name.lastName(),
+    id: faker.datatype.uuid(),
+};
+
+/*
+ *  test cases for registering a user optionally with a password
+ */
+describe('register', () => {
+    test('should register new user with a password', async () => {
+        // user data that the user inputs
+        const textPassword = faker.internet.password();
+        const encryptedPassword = textPassword ? await bcrypt.hash(textPassword, 10) : null;
+        const date = new Date();
+
+        // mock user
+        const user = {
+            canMakeOrgs: false,
+            createdAt: date,
+            email: userData.email,
+            firstName: userData.firstName,
+            lastName: userData.lastName,
+            fullName: userData.firstName + ' ' + userData.lastName,
+            id: userData.id,
+            password: encryptedPassword,
+            preferredLang: 'EN',
+        };
+
+        prismaMock.user.create.mockResolvedValueOnce(user);
+
+        await expect(register(prismaMock, userData, textPassword)).resolves.toEqual({
+            canMakeOrgs: false,
+            createdAt: date,
+            email: userData.email,
+            firstName: userData.firstName,
+            lastName: userData.lastName,
+            fullName: userData.firstName + ' ' + userData.lastName,
+            id: userData.id,
+            password: encryptedPassword,
+            preferredLang: 'EN',
+        });
```

Comment:
nit: I typically like to arrange tests in such a way that there's some setup and then two variables named `output` and `expectedOutput` if possible.  Makes the test very clear and easy to read.